HelpMyStreetFE/HelpMyStreetFE/Services/AddressService.cs

```diff
@@ -184,5 +190,48 @@ public async Task<double> GetDistanceBetweenPostcodes(string postCode1, string p
                 }
             }, $"{CACHE_KEY_PREFIX}-postcode-distances-{postCode1}-{postCode2}", RefreshBehaviour.DontWaitForFreshData, cancellationToken);
         }
+
+        public async Task<double> GetDistanceFromPostcodeForCurrentUser(string postCode, CancellationToken cancellationToken)
+        {
+            var httpContext = _httpContextAccessor.HttpContext;
+            if (httpContext.User != null && httpContext.User.Identity.IsAuthenticated)
+            {
+                var id = int.Parse(httpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value);
+                var user =  await _userRepository.GetUser(id);
+                if (user != null && !String.IsNullOrEmpty("postCode"))
+                {
+                    return await GetDistanceBetweenPostcodes(user.PostalCode, postCode, cancellationToken);
+                }
+            }
+
+            return 0.0;
+        }
+
+        public async Task<LocationWithDistance> GetLocationWithDistance(IEnumerable<IContainsLocation> locationItem, CancellationToken cancellationToken)
```

Comment:
A minor comment, but I would prefer to remove this method and instead do a.First() in the code that calls it.  From the method signature as it stands, one might expect this method to do something more with the IEnumerable than take the first.